simcon_project/users/views/create_assignment.py

```diff
@@ -0,0 +1,156 @@
+        return True
+    return False
+
+@user_passes_test(is_researcher)
+def add_data(request):
+    #Error signs and error messages.
+    errMsg=''
+    success=0
+
+    #Get raw data
+    data=request.POST
+    name=data.get('name')
+    date=data.get('date')
+    researcher=request.user
+    #researcher='researcher@111.com'
+    students=data.get('stuData')
+    templates=data.get('tempData')
+    labels=data.get('labelData')
+
+    #Transfer string type list to list type
+    students=decode(students)
+    templates=decode(templates)
+    labels=decode(labels)
+
+    #Verify date
+    today=datetime.datetime.combine(datetime.date.today(), datetime.time(0, 0))
+    dateTmp=datetime.datetime.strptime(date, "%Y-%m-%d")
+    assignToday=False
+    if dateTmp==today:
+        assignToday=True
+    if dateTmp<today:
+        success=1
+        errMsg=errMsg+'The assigned date cannot before today.\n\n'
+
+    #Save the data that can be directly assigned to assignment.
+    assignment=Assignment()
+    assignment.name=name
+    assignment.date_assigned=date
+    researcher=Researcher.objects.get(email=researcher)
+    assignment.researcher=researcher
+    assignment.save()
+
+    #Determine if students, labels, and templates are empty.
+    stuIsNull=isNull(students)
+    labelIsNull=isNull(labels)
+    tempIsNull=isNull(templates)
+
+    #Assign student information to assignment,
+    #and judge whether there is an error that both students and labels are empty.
+    if stuIsNull:
+        if labelIsNull:
+            success=1
+            errMsg=errMsg+'One of the students or labels must not be empty.\n\n'
+    else:
+        for stu in students:
+            stuTmp=Student.objects.get(email=stu)
+            assignment.students.add(stuTmp)
+
+    #Assign template information to assignment
+    if tempIsNull:
+        success=1
+        errMsg=errMsg+'Template must not be empty.\n\n'
+    else:
+        for temp in templates:
+            tempList=temp.split('--')
+            tempName=tempList[0]
+            tempDate=tempList[1]
+            tempTmp=ConversationTemplate.objects.get(name=tempName,creation_date=tempDate)
+            assignment.conversation_templates.add(tempTmp)
+
+    #Assign label information to assignment
+    if not labelIsNull:
+        for label in labels:
+            labelTmp=SubjectLabel.objects.get(label_name=label)
+            assignment.subject_labels.add(labelTmp)
+
+    #Send email
+    job_defaults = { 'max_instances': 20 }
+    tz = get_localzone()
+    sched = BackgroundScheduler(timezone=tz, job_defaults=job_defaults)
+
+    subject='Simulated Conversation Assignment Update'
+    msg='You received this email because you have a new assignment: '+name+'. Please check the assignment page.'
+    schedId='Assignment--'+name+'--'+date
+    #when an error occurs, there is no need to add this task to the schedule.
+    researcher='simulated.conversation@mail.com'
```

Comment:
change to davids email he has in the settings.py

Reply:
It's in main so should be correct once he brings it up to speed with main